src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -33,86 +34,333 @@ const ActorInit En_Zo_InitVars = {
+void EnZo_DoNothing(EnZo* this, GlobalContext* globalCtx) {
+}
+
+void EnZo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+    s32 pad;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gZoraSkel, NULL, this->jointTable, this->morphTable, 20);
+    EnZo_SetAnimation(&this->skelAnime, 0);
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    this->path = func_8013D648(globalCtx, ENZO_GET_PATH(&this->actor), ENZO_NO_PATH);
+    Actor_SetScale(&this->actor, 0.01f);
+
+    this->actionFunc = EnZo_Walk;
+    this->actor.gravity = -4.0f;
+}
+
+void EnZo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnZo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    EnZo_LookAtPlayer(this, globalCtx);
+    EnZo_PlayWalkingSound(this, globalCtx);
+    EnZo_UpdateCollider(this, globalCtx);
+}
+
+s32 EnZo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                          Gfx** gfx) {
+    EnZo* this = THIS;
+
+    if (limbIndex == 15) {
+        Matrix_InsertTranslation(1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(this->headRot.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->headRot.x, MTXMODE_APPLY);
+        Matrix_InsertTranslation(-1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+
+    if (limbIndex == 8) {
+        Matrix_InsertXRotation_s(-this->upperBodyRot.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->upperBodyRot.x, MTXMODE_APPLY);
+    }
+
+    if ((limbIndex == 8) || (limbIndex == 9) || (limbIndex == 12)) {
+        rot->y += (s16)(Math_SinS(this->limbRotY[limbIndex]) * 200.0f);
+        rot->z += (s16)(Math_CosS(this->limbRotZ[limbIndex]) * 200.0f);
+    }
+    return 0;
+}
+
+void EnZo_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
+    EnZo* this = THIS;
+    Vec3f sp30 = { 400.0f, 0.0f, 0.0f };
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+
+    if (D_8099F578[limbIndex] >= 0) {
+        Matrix_MultiplyVector3fByState(&zeroVec, &this->unk_364[D_8099F578[limbIndex]]);
+    }
+    if (limbIndex == 15) {
+        Matrix_MultiplyVector3fByState(&sp30, &this->actor.focus.pos);
+    }
+    if (limbIndex == 4) {
+        Matrix_MultiplyVector3fByState(&zeroVec, &this->leftFootPos);
+    }
+    if (limbIndex == 7) {
+        Matrix_MultiplyVector3fByState(&zeroVec, &this->rightFootPos);
+    }
+}
+
+Gfx D_8099F5E8[] = {
```

Comment:
This is a material related display list.

Reply:
I've named it based on discussions of the same dlist in the discord.